Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -343,4 +344,26 @@ Value ArrayPrototype::index_of(Interpreter& interpreter)
 
     return Value(-1);
 }
+
+Value ArrayPrototype::reverse(Interpreter& interpreter)
+{
+    auto* array = array_from(interpreter);
+    if (!array)
+        return {};
+
+    if (array->elements().size() == 0)
+        return array;
+
+    auto* array_reverse = Array::create(interpreter.global_object());
+    array_reverse->elements().ensure_capacity(array->elements().size());
+
+    for (ssize_t i = array->elements().size() - 1; i >= 0; --i)
+        array_reverse->elements().append(array->elements().at(i));
+
+    array->elements().clear_with_capacity();
+    array->elements().append(array_reverse->elements());
```

Comment:
No need to create a temporary throwaway `Array` here. Just make a `Vector<Value>` on the stack:

```cpp
Vector<Value> values;
values.ensure_capacity(array->elements.size());
for (ssize_t i = array->elements().size() - 1; i >= 0; --i)
    values.append(array->elements()[i]);
array->elements() = move(values);
```

And yeah, like Linus says, this should be made generic in the future, but an array-only version is okay for now. :)